src/components/SwitzerlandEstimatedCasesDivisionModal.tsx

```diff
@@ -87,26 +88,23 @@ export const SwitzerlandEstimatedCasesDivisionModal = ({
   show,
   handleClose,
 }: Props) => {
-  const [caseCounts, setCaseCounts] = useState<CaseCountEntry[] | undefined>();
+  //const [caseCounts, setCaseCounts] = useState<CaseCountEntry[] | undefined>();
   const [showSwissRegions, setShowSwissRegions] = useState(true);
-
   const { dateFrom, dateTo, country } = variantSampleSet.sampleSelector;
-  useEffect(() => {
-    let isSubscribed = true;
-    const controller = new AbortController();
-    const signal = controller.signal;
 
-    getCaseCounts({ dateFrom, dateTo, country }, true, signal).then(caseCounts => {
-      if (isSubscribed) {
-        setCaseCounts(caseCounts);
-      }
-    });
+  const fetchCaseCounts = async () => {
+    const data = await getCaseCounts({ dateFrom, dateTo, country }, true, new AbortController().signal);
+    return data;
+  };
 
-    return () => {
-      isSubscribed = false;
-      controller.abort();
-    };
-  }, [dateFrom, dateTo, country]);
+  const { isLoading, isSuccess, error, isError, data: caseCounts, refetch } = useQuery<
```

Comment:
We can use these properties returned from `useQuery` to check if the request `isLoading`, `isSuccess` or `isError`, which catches any error in the request.